tempesta_fw/t/unit/test_tfw_str.c

```diff
@@ -1178,6 +1178,292 @@ TEST(tfw_str_collect_cmp, collect_chunks)
 	EXPECT_EQ(TFW_STR_CHUNKN(&out), 5);
 }
 
+TEST(tfw_str_add_compound, plain)
+{
+	TfwStr *s;
+	TfwStr s1 = {
+		.len = 16,
+		.ptr = "abcdefghijklmnop"
+	};
+	TfwStr s2 = {
+		.len = 10,
+		.ptr = "qrstuvwxyz"
+	};
+
+	s = tfw_str_add_compound(str_pool, &s1);
+	*s = s2;
+	EXPECT_TRUE(TFW_STR_CHUNKN(&s1) == 2);
+	EXPECT_TRUE(tfw_str_eq_cstr(&s1, "abcdefghijklmnopqrstuvwxyz",
+				    sizeof("abcdefghijklmnopqrstuvwxyz") - 1,
+				    0));
+}
+
+TEST(tfw_str_add_compound, compound)
+{
+	TfwStr *s;
+	int chunks;
+	TFW_STR(s1, "abcdefghijklmnop");
+	TfwStr s2 = {
+		.len = 10,
+		.ptr = "qrstuvwxyz"
+	};
+
+	chunks = TFW_STR_CHUNKN(s1);
+
+	s = tfw_str_add_compound(str_pool, s1);
+	*s = s2;
+	EXPECT_TRUE(TFW_STR_CHUNKN(s1) == chunks + 1);
+	EXPECT_TRUE(tfw_str_eq_cstr(s1, "abcdefghijklmnopqrstuvwxyz",
+				    sizeof("abcdefghijklmnopqrstuvwxyz") - 1,
+				    0));
+}
+
+TEST(tfw_str_add_duplicate, both_plain)
+{
+	TfwStr *s, *dup, *dup_end;
+	TfwStr s1 = {
+		.len = 16,
+		.ptr = "abcdefghijklmnop"
+	};
+	TfwStr s2 = {
+		.len = 16,
+		.ptr = "abcdefghijklmnop"
+	};
+
+	s = tfw_str_add_duplicate(str_pool, &s1);
+	*s = s2;
+	EXPECT_TRUE(TFW_STR_DUP(&s1));
+	TFW_STR_FOR_EACH_DUP(dup, &s1, dup_end) {
+		EXPECT_TRUE(tfw_str_eq_cstr(dup, "abcdefghijklmnop",
```

Comment:
Need to use different values for duplicates. If the function has a bug when both duplicates - pointer to the same `TfwStr` instance, how it can be found?